pkg/containertools/dockerfilegenerator.go

```diff
@@ -8,14 +8,16 @@ import (
 )
 
 const (
-	defaultBinarySourceImage = "quay.io/operator-framework/upstream-opm-builder"
-	DefaultDbLocation        = "/database/index.db"
-	DbLocationLabel          = "operators.operatorframework.io.index.database.v1"
+	defaultBinarySourceImage    = "quay.io/operator-framework/upstream-opm-builder"
+	DefaultDbLocation           = "/database/index.db"
+	DefaultConfigFolderLocation = "/configs/"
+	DbLocationLabel             = "operators.operatorframework.io.index.database.v1"
+	ConfigsLocationLabel        = "operators.operatorframework.io.configs.v1"
 )
 
 // DockerfileGenerator defines functions to generate index dockerfiles
 type DockerfileGenerator interface {
-	GenerateIndexDockerfile(string, string) string
+	GenerateIndexDockerfile(string, string, string) string
```

Comment:
Ref: https://github.com/operator-framework/enhancements/blob/master/enhancements/declarative-index-config.md#migration-plan

When the declarative index feature is delivered we'll just be generating an index with only a config path. This change is only a stop gap measure so that master is not broken while new features are being developed. It should be changed back to accept only one before the release of the new binary.

Reply:
Does it make more sense for us to create some new interfaces rather than modify existing ones? It would probably make it more obvious so that we know what needs to get cleaned up afterwards

Reply:
I second what Kevin is saying. I prefer we define the interfaces we want up front.

Reply:
The reason I'm having trouble understanding the request for a new interface is because ultimately, we do need the same `DockerfileGenerator` interface even when we will only include the configs and not the database. 

But, that will be possible only when the ability to serve the apis with the configs before available. Ref: https://github.com/operator-framework/enhancements/blob/master/enhancements/declarative-index-config.md#usage-of-the-json-representation-of-packages-by-olm-components
In which case we generate the Dockerfile in a way that only adds the configs and serves those configs. 

As for how it stands now, if we have an implementation for `DockerfileGenerator` that adds the configs (but not the database), the container image will not have the ability to serve the apis. Therefore the database must be included (and served) in the container.

Reply:
This may have been discussed more in the last week, but it seems like we'll have the config server done and merged in the next week or two. I would agree with Kevin and Nick that it seems like it will be unnecesssary for us to build images that contain both databases and configs.